.github/workflows/s4hana-build-connector-template.yml

```diff
@@ -0,0 +1,96 @@
+name: Build (S/4 HANA Connector)
+
+on:
+  workflow_call:
+    inputs:
+      hana-connector-group:
+        required: true
+        type: string
+
+jobs:
+  build-examples:
+    name: Build Examples
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set Up Ballerina
+        uses: ballerina-platform/setup-ballerina@v1.1.0
+        with:
+          version: latest
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 17.0.7
+
+      - name: Build the Package
+        env:
+          packageUser: ${{ github.actor }}
+          packagePAT: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ./gradlew :${{ inputs.hana-connector-group }}-examples:build :${{ inputs.hana-connector-group }}-sanitation:build -x :${{ inputs.hana-connector-group }}-ballerina:build
+
+  generate-matrix:
+    name: Generate Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      directories: ${{ steps.set-matrix.outputs.directories }}
+    steps:
+      - name: Checkout the Repository
+        uses: actions/checkout@v3
+      - name: Set matrix for build
+        id: set-matrix
+        run: |
+          folders=()
+          cd ballerina
+          for dir in $(find . -type d -maxdepth 1  -mindepth 1 | cut -c3-); do
+            if [[ "$dir" == resources ]]; then
+              continue
+            fi
+            folders+=("$dir"); 
+          done
+          echo "directories=$(jq -nc --args '$ARGS.positional' ${folders[@]})" >> $GITHUB_OUTPUT
+
+  build-connectors:
+    name: Build Connectors
+    runs-on: ubuntu-latest
+    needs: generate-matrix
+    strategy:
+      matrix: 
+        directory: ${{ fromJSON(needs.generate-matrix.outputs.directories) }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 17.0.7
+
+      - name: Build the Package
+        env:
+          packageUser: ${{ github.actor }}
+          packagePAT: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ./gradlew :${{ inputs.hana-connector-group }}-ballerina:${{ matrix.directory }}:build -x test -PbuildUsingDocker=nightly
+          ./gradlew :${{ inputs.hana-connector-group }}-ballerina:${{ matrix.directory }}:test -PbuildUsingDocker=nightly
+
+      - name: Upload coverage reports
+        uses: actions/upload-artifact@v3
+        with:
+          path: ballerina/**/coverage-report.xml
+
+  upload-codcov:
+    name: Upload Codecov
+    runs-on: ubuntu-latest
+    needs: build-connectors
+    steps:
+      - name: Download coverage reports
+        uses: actions/download-artifact@v3
+
+      - name: Generate Codecov Report
+        uses: codecov/codecov-action@v3
```

Comment:
I think we should use the token here for avoid failing to upload the report.

Reply:
Hmm, this action by default looks for "secrets.CODECOV_TOKEN" for this. If we have used the same name. We dont need to explicitly mention it here.

Reply:
Ack.